src/app/custom-editor/custom-editor.component.ts

```diff
@@ -136,24 +137,29 @@ export class CustomEditorComponent implements OnInit, OnDestroy, OnChanges {
     this.currentView = editorViewsConst[index];
     if (this.currentView == 'raw') {
       this.editorContent = this.rawFile;
-    }
-    if (this.currentView == 'xml') {
+    } else {
       this.prettify();
     }
   }
 
   prettify(): void {
     if (this.editorContent && !this.isPrettified) {
-      prettier
-        .format(this.editorContent, {
-          parser: 'html',
-          plugins: [prettierPluginHtml],
-          bracketSameLine: true,
-        })
-        .then((result: string): void => {
-          this.setValue(result);
-          this.isPrettified = true;
-        });
+      if (this.currentView == 'xml' && this.contentType == 'xml') {
+        prettier
+          .format(this.editorContent, {
+            parser: 'html',
+            plugins: [prettierPluginHtml],
+            bracketSameLine: true,
+          })
+          .then((result: string): void => {
+            this.setValue(result);
+            this.isPrettified = true;
+          });
+      }
+      if (this.currentView == 'json' && this.contentType == 'json') {
+        this.editorContent = JSON.stringify(JSON.parse(this.editorContent), null, '  '); //Indents set to two spaces
```

Comment:
You could create a constant for the spaces and make it readable without comment:
```suggestion
        this.editorContent = JSON.stringify(JSON.parse(this.editorContent), null, this.INDENT_TWO_SPACES);
```

Reply:
Fixed